scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryImportResource.java

```diff
@@ -0,0 +1,259 @@
+      repository = manager.create(
+        repository,
+        pullChangesFromRemoteUrl(request)
+      );
+      eventBus.post(new RepositoryImportEvent(HandlerEventType.MODIFY, repository, false));
+
+      return Response.created(URI.create(resourceLinks.repository().self(repository.getNamespace(), repository.getName()))).build();
+    } catch (Exception e) {
+      eventBus.post(new RepositoryImportEvent(HandlerEventType.MODIFY, repository, true));
+      throw e;
+    }
+  }
+
+  @VisibleForTesting
+  Consumer<Repository> pullChangesFromRemoteUrl(RepositoryImportDto request) {
+    return repository -> {
+      try (RepositoryService service = serviceFactory.create(repository)) {
+        PullCommandBuilder pullCommand = service.getPullCommand();
+        if (!Strings.isNullOrEmpty(request.getUsername()) && !Strings.isNullOrEmpty(request.getPassword())) {
+          pullCommand
+            .withUsername(request.getUsername())
+            .withPassword(request.getPassword());
+        }
+
+        pullCommand.pull(request.getImportUrl());
+      } catch (IOException e) {
+        throw new InternalRepositoryException(repository, "Failed to import from remote url", e);
+      }
+    };
+  }
+
+  /**
+   * Check repository type for support for the given command.
+   *
+   * @param type    repository type
+   * @param cmd     command
+   * @param request request object
+   */
+  private void checkSupport(Type type, Command cmd, Object request) {
+    if (!(type instanceof RepositoryType)) {
+      logger.warn("type {} is not a repository type", type.getName());
+
+      throw new WebApplicationException(Response.Status.BAD_REQUEST);
+    }
+
+    Set<Command> cmds = ((RepositoryType) type).getSupportedCommands();
+
+    if (!cmds.contains(cmd)) {
+      logger.warn("type {} does not support this type of import: {}",
+        type.getName(), request);
+
+      throw new WebApplicationException(Response.Status.BAD_REQUEST);
+    }
+  }
+
+  private Type type(String type) {
+    RepositoryHandler handler = manager.getHandler(type);
+
+    if (handler == null) {
+      logger.warn("no handler for type {} found", type);
+
+      throw new WebApplicationException(Response.Status.NOT_FOUND);
+    }
+
+    return handler.getType();
+  }
+
+  @Getter
+  @Setter
+  @NoArgsConstructor
+  @SuppressWarnings("java:S2160")
+  public static class RepositoryImportDto extends RepositoryDto implements ImportRepositoryDto {
+    private String importUrl;
+    private String username;
+    private String password;
+
+    RepositoryImportDto(Links links, Embedded embedded) {
+      super(links, embedded);
+    }
+  }
+
+  interface ImportRepositoryDto {
+
+    String getNamespace();
+    @Pattern(regexp = ValidationUtil.REGEX_REPOSITORYNAME)
+    String getName();
+    @NotEmpty
+    String getType();
+    @Email
+    String getContact();
+    String getDescription();
+    List<HealthCheckFailureDto> getHealthCheckFailures();
+    Instant getLastModified();
```

Comment:
I don't think that HealthCheckFailures nor lastModified should be part of the import dto